integrationm/common/com/cultofbits/customizations/audit/AuditInstanceUpdater.groovy

```diff
@@ -0,0 +1,134 @@
+package com.cultofbits.customizations.audit
+
+import com.cultofbits.integrationm.service.actionpack.RecordmActionPack
+import com.google.common.cache.Cache
+import com.cultofbits.integrationm.service.dictionary.recordm.Definition
+import com.cultofbits.integrationm.service.dictionary.recordm.FieldDefinition
+import com.cultofbits.integrationm.service.dictionary.recordm.RecordmMsg
+import com.google.common.cache.CacheBuilder
+import org.json.JSONObject
+
+class AuditInstanceUpdater {
+
+
+    static Definition getDefinitionFromCache(cacheOfAuditFieldsForDefinition, RecordmMsg msg, RecordmActionPack recordm){
+        Definition definition = cacheOfAuditFieldsForDefinition.get(msg.type, { recordm.getDefinition(msg.getDefinitionName()).getBody() })
+
+        if (definition.getVersion() != msg.getDefinitionVersion()) {
+            cacheOfAuditFieldsForDefinition.invalidate(msg.type)
+            definition = cacheOfAuditFieldsForDefinition.get(msg.type, { recordm.getDefinition(msg.getDefinitionName()).getBody() })
+        }
+
+        return definition
+    }
+
+    static Cache<String,Definition> NewCache(int maximumSize){
+        return CacheBuilder.newBuilder()
+                .maximumSize(maximumSize)
+                .build()
+    }
+
+    /**
+     * @return todos os campos da definiçao @definitionName com a descrição
+     *  [$]audit.(creator|updater).(username|uri|time)
+     * O vetor devolvido tem em cada entrada o tuplo (fieldId: fieldId, name:fieldName, op : (creator ou updater), args: (username ou uri ou time))
+     * */
+    def static getAuditFields(log, RecordmMsg msg, Definition definition) {
```

Comment:
Make `log` the last argument, also, this method can call the cache, you don't need to provide the definition. It should be enough `getAuditFields(RecordmMsg msg, log)`